make/photon/registry/builder

```diff
@@ -1,44 +1,32 @@
 #!/bin/bash
 
-set +e
+set -e
 
 if [ -z $1 ]; then
-  error "Please set the 'version' variable"
+  echo "Please set the 'version' variable"
   exit 1
 fi
 
 VERSION="$1"
 
-set -e
-
 # the temp folder to store binary file...
 mkdir -p binary
 rm -rf binary/registry || true
 
 cd `dirname $0`
-cur=$PWD
 
 # the temp folder to store distribution source code...
 TEMP=`mktemp -d ${TMPDIR-/tmp}/distribution.XXXXXX`
-git clone -b $VERSION https://github.com/distribution/distribution.git $TEMP
+git clone -b $VERSION --depth 1 https://github.com/distribution/distribution.git $TEMP
 
 # add patch redis
 cd $TEMP
-git apply $cur/redis.patch
-cd $cur
+git apply ~-/redis.patch
```

Comment:
does it support all the linux OS versions?

Reply:
Yes.
It is supported by all currently used bash.